packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -131,6 +131,80 @@ enum BottomNavigationBarType {
 /// ```
 /// {@end-tool}
 ///
+/// {@tool dartpad --template=stateful_widget_material}
+/// This example shows [BottomNavigationBar] used in a [Scaffold] Widget with
+/// different interaction patterns. Tapping on the first [BottomNavigationBarItem]
+/// uses the [ScrollController] to animate the [ListView] to the top. The second
+/// [BottomNavigationBarItem] opens a [DropdownButton].
+///
+///```dart
+/// int _selectedIndex = 0;
+/// final ScrollController _homeController = ScrollController();
+/// String dropdownValue = 'One';
+/// FontWeight _fontWeight = FontWeight.normal;
+///
+/// void _onItemTapped(int index) {
+///   if (index == 0) {
+///     _homeController.animateTo(
+///         0.0,
+///         duration: Duration(milliseconds: 500),
+///         curve: Curves.easeOut
+///     );
+///   }
+///   else{
+///     setState(() {
+///       _fontWeight = index == 1 ? FontWeight.bold : FontWeight.normal;
+///       _selectedIndex = index;
+///     });
+///   }
+/// }
+///
+/// Widget _listViewBody() {
+///   return ListView.separated(
+///       controller: _homeController,
+///       itemBuilder: (context, index) {
+///         return Center(
+///             child: Text(
+///               "Item $index",
+///               style: TextStyle(fontWeight: _fontWeight),
```

Comment:
You could just do this:
```suggestion
///               style: TextStyle(
///                 fontWeight: _selectedIndex == 1 ? FontWeight.bold : FontWeight.normal,
///.              ),
```
Then you won't need to save `_fontWeight` in your state.